src/copaw/app/routers/skills.py

```
@@ -126,12 +140,18 @@ async def install_from_hub(request: HubInstallRequest):
         raise HTTPException(status_code=400, detail=str(e)) from e
     except RuntimeError as e:
         # Upstream hub is flaky/rate-limited sometimes; surface as bad gateway.
-        raise HTTPException(status_code=502, detail=str(e)) from e
+        detail = str(e) + _github_token_hint(request.bundle_url)
+        logger.exception(
+            "Skill hub install failed (upstream/rate limit): %s",
+            e,
```

Ruff (0.15.2)

[warning] Redundant exception object included in `logging.exception` call

(TRY401)